source/qx/io/jsonrpc/message/Request.js

```diff
@@ -1,62 +0,0 @@
-qx.Class.define("qx.io.jsonrpc.message.Request",{
-  extend: qx.io.jsonrpc.Notification,
-  statics: {
-    /**
-     * Static counter for all request ids
-     */
-    __current_request_id : 0,
-
-    getCurrentId() {
-      return qx.io.jsonrpc.message.Request.__current_request_id;
-    },
-
-    resetId() {
-      qx.io.jsonrpc.message.Request.__current_request_id = 0;
-    }
-  },
-  properties: {
-    /**
-     * The integer id of the request
-     */
-    id : {
-      check: value => qx.lang.Type.isNumber(value) && parseInt(value) === value
-    },
-    /**
-     * Promise that is resolved when the response to this request has been received
-     */
-    promise: {
-      check: "qx.Promise"
```

Comment:
The idea is that you can pass around message objects that are in a batch and have them resolve in places other than where the Client method is used to send the batch, for example, if a periodic job collects the messages. I am not sure if we really need a qx property for that, it could also just be a getter for a private variable.